### ISSUE TITLE
no-floating-promises recognized synchronous function "getModule" of "vuex-module-decorators" as returning promise one

### ISSUE THREAD

**Issue by TokugawaTakesi** [typescript-eslint/typescript-eslint#985]
**Repro**

I attached my project. Once `npm install` done, run `eslint Source/ProjectInitializer/PageComponents/ProjectFilesAndFoldersIsBeingGeneratedNow.ts` in the project directory (or, if you use IntelliJ IDEA family IDE, you will see ESLint violation highlight in this file).

[📁 SourceFiles ](https://github.com/typescript-eslint/typescript-eslint/files/3624993/repro.zip)

```JSON
{
  "rules": {
    "@typescript-eslint/no-floating-promises": "error"
  }
}
```

```TS
@Component({
  template,
  components: { VariableLengthArcLoadingSpinner }
})
export default class ProjectFilesAndFoldersIsBeingGeneratedNow extends Vue {

  private packageJsonGenerationDone: boolean = false;
  private hikariConfigYamlGenerationDone: boolean = false;

  private created(): void {
    InterProcessFacilitator.onProjectPreparationDone = () => {
      getModule(CommonStoreModule).proceedToNextStep();
    };
  }
}
```

**Expected Result**
No ESLint violation in ` getModule(CommonStoreModule).proceedToNextStep();` line.

**Actual Result**

![image](https://user-images.githubusercontent.com/41653501/65125505-6dfb9380-da30-11e9-8c76-75a50a029a60.png)

**Additional Info**
No Promises in declarations for `getModule` function:

```
export declare function getModule<M extends VuexModule>(moduleClass: ConstructorOf<M>, store?: Store<any>): M;
```

I did not find the full file in repository of [vuex-module-decorators](https://github.com/championswimmer/vuex-module-decorators), so I'll paste it here: 

```typescript
import { ActionTree, GetterTree, Module as Mod, ModuleTree, MutationTree, Store, ActionContext } from 'vuex';
export declare class VuexModule<S = ThisType<any>, R = any> implements Mod<S, R> {
    static namespaced?: boolean;
    static state?: any | (() => any);
    static getters?: GetterTree<any, any>;
    static actions?: ActionTree<any, any>;
    static mutations?: MutationTree<any>;
    static modules?: ModuleTree<any>;
    modules?: ModuleTree<any>;
    namespaced?: boolean;
    getters?: GetterTree<S, R>;
    state?: S | (() => S);
    mutations?: MutationTree<S>;
    actions?: ActionTree<S, R>;
    context: ActionContext<S, R>;
    constructor(module: Mod<S, any>);
}
declare type ConstructorOf<C> = {
    new (...args: any[]): C;
};
export declare function getModule<M extends VuexModule>(moduleClass: ConstructorOf<M>, store?: Store<any>): M;
export {};

```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.3.0` |
| `@typescript-eslint/parser`        | `2.3.0` |
| `TypeScript`                       | `3.6.3` |
| `ESLint`                           | `6.4.0` |
| `node`                             | `10.15.3` |
| `npm`                              | `6.11.3` |

**Comment by bradzacher** [comment]
You are misunderstanding the error message, because your IDE is incorrectly reporting the error's position.

![image](https://user-images.githubusercontent.com/7462525/65169293-18f15a80-d9fb-11e9-9207-185e2e47cd07.png)

It is in fact your `proceedToNextStep` function which returns a promise, which needs to be handled.

**Comment by TokugawaTakesi** [comment]
You right. My mistake. Very sorry.
